s3fs/core.py

```diff
@@ -1814,14 +1814,58 @@ def __init__(
         self.append_block = False
 
         if "a" in mode and s3.exists(path):
-            loc = s3.info(path)["size"]
+            # See:
+            #  put: https://boto3.amazonaws.com/v1/documentation/api/latest
+            #  /reference/services/s3.html#S3.Client.put_object
+
+            # head : https://boto3.amazonaws.com/v1/documentation/api/latest
+            # /reference/services/s3.html#S3.Client.head_object
+            head = {
+                k: v
+                for k, v in self._call_s3(
+                    "head_object",
+                    self.kwargs,
+                    Bucket=bucket,
+                    Key=key,
+                    **version_id_kw(version_id),
+                    **self.req_kw,
+                ).items()
+                if (
+                    k
+                    in {
+                        "CacheControl",
+                        "ContentDisposition",
+                        "ContentEncoding",
+                        "ContentLanguage",
+                        "ContentLength",
+                        "ContentType",
+                        "Expires",
+                        "WebsiteRedirectLocation",
+                        "ServerSideEncryption",
+                        "SSECustomerAlgorithm",
+                        "SSEKMSKeyId",
+                        "BucketKeyEnabled",
+                        "StorageClass",
+                        "ObjectLockMode",
+                        "ObjectLockRetainUntilDate",
+                        "ObjectLockLegalHoldStatus",
+                        "Metadata",
```

Comment:
Let's put these into some constant to simplify the logic here, as there are currently too many levels.